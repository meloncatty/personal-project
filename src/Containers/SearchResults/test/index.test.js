import React from 'react'
import { shallow, mount } from 'enzyme'
import { SearchResults } from '../index.js'

describe('SearchResults', () => {
  it('should match snapshots', () => {
    const resultsSuccess = [
      {
        id: 1,
        title: 'Test the things'
      }
    ]
  const wrapper = shallow(
    <SearchResults nextPageSuccess={[]} resultsSuccess={resultsSuccess}/>
  )

  expect(wrapper).toMatchSnapshot()
  })

  it("should have default state", () => {
    const resultsSuccess = [
      {
        id: 1,
        title: 'Test the things'
      }
    ]
    const wrapper = shallow(<SearchResults nextPageSuccess={[]} resultsSuccess={resultsSuccess} />)
    const expected = {
      redirectToArticle: false,
      canUserPost: false,
      pageCounter: 1
    }

    expect(wrapper.state()).toEqual(expected)
  })

  describe('redirectToArticle', () => {
    it("should update state when article is clicked", () => {
      const resultsSuccess = [
        {
          id: 1,
          title: 'Test the things',

        }
      ]
      const fetchNextPage = jest.fn()
      const wrapper = mount(<SearchResults fetchNextPage={fetchNextPage} resultsTotalHits={100} nextPageSuccess={resultsSuccess} resultsSuccess={resultsSuccess} />)
      wrapper.instance().cleanQueryResults
      wrapper.find('button').simulate('click')

      expect(wrapper.state('pageCounter')).toEqual(2)
    })

    it("should call fetchNextPage when clicked", () => {
      const resultsSuccess = [
        {
          id: 1,
          title: 'Test the things',

        }
      ]
      const fetchNextPage = jest.fn()
      const wrapper = mount(<SearchResults fetchNextPage={fetchNextPage} resultsTotalHits={100} nextPageSuccess={resultsSuccess} resultsSuccess={resultsSuccess} />)
      wrapper.instance().cleanQueryResults
      wrapper.find('button').simulate('click')

      expect(wrapper.fetchNextPage).toHaveBeenCalled
    });
  })

  describe("displayErrorText", () => {
    it("should display error text when errors occur", () => {
      const resultsSuccess = [
        {
          id: 1,
          title: 'Test the things',

        }
      ]
      const fetchNextPage = jest.fn()
      const wrapper = mount(<SearchResults resultsHaveErrored={true} nextPageErrored={true} nextPageSuccess={[]} resultsSuccess={resultsSuccess} />)

      expect(wrapper.instance().dispalyErrorText).toHaveBeenCalled
    })
  })

  describe("cleanQueryRestuls", () => {
    const resultsSuccess = [
      {
        id: 1,
        title: 'Test the things',

      }
    ]
    const fetchNextPage = jest.fn()
    const wrapper = mount(<SearchResults userAuthentication={[0]} fetchNextPage={fetchNextPage} nextPageSuccess={[]} resultsSuccess={resultsSuccess} />)
    wrapper.instance().cleanQueryResults
    wrapper.find('.archive-denied').simulate('click')

    expect(wrapper.fetchNextPage).toHaveBeenCalled
  });
})
