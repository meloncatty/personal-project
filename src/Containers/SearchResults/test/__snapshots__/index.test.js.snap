// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults should match snapshots 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchResults
    nextPageSuccess={Array []}
    resultsSuccess={
        Array [
            Object {
              "id": 1,
              "title": "Test the things",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="results-list"
>
          
</section>,
        undefined,
      ],
      "className": "results-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            Array [],
            undefined,
            false,
          ],
          "className": "results-list",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          false,
        ],
        "type": "section",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="results-list"
>
            
</section>,
          undefined,
        ],
        "className": "results-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              Array [],
              undefined,
              false,
            ],
            "className": "results-list",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            false,
          ],
          "type": "section",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
